src/services/github.ts

```diff
@@ -181,38 +183,53 @@ const createFileContent = async (folderName, fileName, base64, commitMsg) => {
   }
 }
 
-const changeHeadsCommitSha = async commitSha => {
-  const endpoint = `${ENDPOINT_BASE}/git/refs/heads/master`
-  const bodyData = {
-    sha: commitSha,
-  }
-  try {
-    const result = await fetch(endpoint, {
-      method: 'POST',
-      headers: {
-        Authorization: `token ${GITHUB_API_TOKEN}`,
-        'Content-Type': 'application/json',
-      },
-      body: JSON.stringify(bodyData),
-    })
-    const data = await result.json()
+type Base64ExtType = {
+  imageExtension: string
+  base64: string
+}
 
-    return { data: data, error: !result.ok }
-  } catch (err) {
-    console.error(`Error requesting tree sha`, err)
-    return { error: true }
+type LinksMetadadaType = {
+  community?: Array<Record<string, unknown>>
+  documentation?: Array<Record<string, unknown>>
+}
+
+type GardenMetadataType = {
+  address: string
+  name: string
+  description: string
+  forum: string
+  links?: LinksMetadadaType
+  logo?: Base64ExtType
+  logo_type?: Base64ExtType
+  token_logo?: Base64ExtType
+}
+
+type GardenMetadataGithubType = Omit<
+  GardenMetadataType,
+  'logo' | 'logo_type' | 'token_logo'
+> & {
+  logo: string | null
+  logo_type: string | null
+  token_logo: string | null
+}
```

Comment:
its the way extend some type with overrides some props using omit.

That way we dont duplicate code and keep consistency.

Reply:
Why do you need to override? Why don't type as `Base64ExtType | string | null` ?

Reply:
Nice question @tmoutinho. The main reason is because GardenMetadata and GardenMetadataGithub are 2 different objects used to different proposols. So its easy type objects and keep them separated it could avoid misuse and alot if's checking for typing and confusion.

Having different types on return functions and params allow fast understand what type object is needed or its backing.

Using that suggestion will work? Yes will. But will need more effort to keep and update on the future, immo.

Reply:
Keep and update in the future? I don't understand what you mean by that. Can you clarify please?